rpc_proxy.go

```diff
@@ -0,0 +1,246 @@
+package main
+
+import (
+	"context"
+	"crypto/x509"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+	"time"
+
+	"github.com/btcsuite/btcutil"
+	"github.com/improbable-eng/grpc-web/go/grpcweb"
+	grpcProxy "github.com/mwitkow/grpc-proxy/proxy"
+	log "github.com/sirupsen/logrus"
+	"github.com/tdex-network/tdex-daemon/pkg/macaroons"
+	"github.com/tdex-network/tdex-daemon/pkg/tdexdconnect"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/backoff"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/metadata"
+	"gopkg.in/macaroon.v2"
+)
+
+const (
+	defaultConnectTimeout = 15 * time.Second
+	defaultServerTimeout  = 10 * time.Second
+	defaultServerAddr     = ":3030"
+)
+
+var (
+	// maxMsgRecvSize is the largest message our REST proxy will receive. We
+	// set this to 200MiB atm.
+	maxMsgRecvSize = grpc.MaxCallRecvMsgSize(1 * 1024 * 1024 * 200)
+	connectUrlPath = filepath.Join(
+		btcutil.AppDataDir("tdex-dashboard", false), "tdexdconnect.txt",
```

Comment:
better to have a JSON file we use a config, so we can dump the current localStorage there cc @Janaka-Steph

Reply:
so this means we should dump it every time it is affected by a change to keep it in sync, shouldn't it?
We want just the connect URL, maybe we could dump just the settings, but there should be some prop that can be changed by the user if I'm not wrong, so the problem would still persist. 

Do we need to use anything more than the connect URL? If not, I would dump just that prop to a file since that's not something the user is allowed to change unless logging out and pairing with a new daemon

Reply:
Ideally we will allow from the same app to control multiple daemons (ie. multiple connect URLs) that's why I think a structured format to store data is a better start.

Maybe I am wrong, but using localStorage API in Tauri is already like saving on file system, not on webview?

Reply:
Tauri FS writeFile() should write a stringified JSON file in a text file https://tauri.studio/en/docs/api/js/modules/fs/#writefile

Reply:
ok, but the current localStorage we are using on JS side, when run in Tauri, will be kept inside the webview memory or it is persisted on filesystem? Or we need ourself to manually have a solution for persistence in Tauri?

Reply:
redux-persist has been tweaked to use Tauri FS instead of LocalStorage

Reply:
exactly, so will most likely be a JSON on the filesystem? makes sense to have the tdexdconncet field there

Reply:
(keep in mind will likley become an array of URLs object soon)

Reply:
One file per reducer seems fine. Stringified json in text file so it's easy to JSON.parse(). 
Currently the only persisted reducer is `Settings`.